src/structures/CommandInteractionOptionResolver.js

```diff
@@ -35,14 +28,24 @@ class CommandInteractionOptionResolver {
      * @private
      */
     this._subCommand = null;
-    if (this._options[0]?.type === 'SUB_COMMAND_GROUP') {
-      this._group = this._options[0].name;
-      this._options = this._options[0].options ?? [];
+
+    let data = options;
+    if (data[0]?.type === 'SUB_COMMAND_GROUP') {
+      this._group = data[0].name;
+      data = data[0].options ?? [];
     }
-    if (this._options[0]?.type === 'SUB_COMMAND') {
-      this._subCommand = this._options[0].name;
-      this._options = this._options[0].options ?? [];
+    if (data[0]?.type === 'SUB_COMMAND') {
+      this._subCommand = data[0].name;
+      data = data[0].options ?? [];
     }
+
+    /**
+     * The interaction options array.
+     * @name CommandInteractionOptionResolver#data
+     * @type {ReadonlyArray<CommandInteractionOption>}
+     * @readonly
+     */
+    Object.defineProperty(this, 'data', { value: Object.freeze(data) });
```

Comment:
Should this be modified to have `data` be the top-level options array? For cases where people want to access the original interaction option array with the sub-command included.

Reply:
@iCrawl @monbrey

Reply:
🤔 Excellent question. I personally lean to yes, I'd like to see discord.js providing the completely original, unparsed data.

This might mean you keep a private `_data` / `_options` property too if you want to do keep the hoisted sub-command options somewhere.

Reply:
Per [discussion](https://discord.com/channels/222078108977594368/682166281826598932/867182323316424774), this will stay as-is in this PR.